homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -0,0 +1,182 @@
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    DATA_SURE_PETCARE,
+    DEFAULT_DEVICE_CLASS,
+    SPC,
+    TOPIC_UPDATE,
+    SureLocationID,
+    SureLockStateID,
+    SureThingID,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up Sure PetCare Flaps sensors based on a config entry."""
+    if discovery_info is None:
+        return
+
+    entities = []
+
+    spc = hass.data[DATA_SURE_PETCARE][SPC]
+
+    for thing in spc.ids:
+        sure_id = thing[CONF_ID]
+        sure_type = thing[CONF_TYPE]
+
+        if sure_type == SureThingID.FLAP.name:
+            entity = Flap(sure_id, thing[CONF_NAME], spc)
+        elif sure_type == SureThingID.PET.name:
+            entity = Pet(sure_id, thing[CONF_NAME], spc)
+
+        entities.append(entity)
+
+    async_add_entities(entities, True)
+
+
+class SurePetcareBinarySensor(BinarySensorDevice):
+    """A binary sensor implementation for Sure Petcare Entities."""
+
+    def __init__(
+        self, _id: int, name: str, spc, device_class: str, sure_type: SureThingID
+    ):
+        """Initialize a Sure Petcare binary sensor."""
+        self._id = _id
+        self._name = name
+        self._spc = spc
+        self._device_class = device_class
+        self._sure_type = sure_type
+        self._state = {}
+
+    @property
+    def is_on(self):
+        """Return true if entity is on/unlocked."""
+        return bool(self._state)
+
+    @property
+    def should_poll(self):
+        """Return true."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEFAULT_DEVICE_CLASS if not self._device_class else self._device_class
+
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return f"{self._spc.household_id}-{self._id}"
+
+    async def async_update(self):
+        """Get the latest data and update the state."""
+        self._state = self._spc.states[self._sure_type][self._id].get("data")
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+
+        @callback
+        def update():
+            """Update the state."""
+            self.async_schedule_update_ha_state(True)
+
+        # pylint: disable=attribute-defined-outside-init
+        self._async_unsub_dispatcher_connect = async_dispatcher_connect(
+            self.hass, TOPIC_UPDATE, update
+        )
+
+    async def async_will_remove_from_hass(self):
+        """Disconnect dispatcher listener when removed."""
+        # pylint: disable=using-constant-test
```

Comment:
I think we can remove this pylint disable. Not sure why it's here.

Reply:
Because the Azure pipeline told me so :D but removing it now seems okay :)